qutebrowser/completion/models/histcategory.py

```diff
@@ -111,6 +125,8 @@ def set_pattern(self, pattern):
             message.error("Error with SQL query: {}".format(e.text()))
             return
         self.setQuery(self._query.query)
+        if self.rowCount() == 0 and not self.canFetchMore():
```

Comment:
You could do 

```suggestion
        if self.rowCount() and not self.canFetchMore():
```

but no strong feelings either way.